lib/src/models/flags.dart

```diff
@@ -0,0 +1,13 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'flags.freezed.dart';
+
+@freezed
+class Flags with _$Flags {
+  const Flags._();
+
+  const factory Flags({
+    required bool isDebug,
+    required bool isAppXS,
+  }) = Flags_mode;
+}
```

Comment:
Some minor things here:
a) I'd say it's best to still make it a `sealed class`
b) When there's a single (unnamed) factory, you should name the generated factory the same as the class, but private. So, in this case, instead of `Flags_mode` it should just be `_Flags`.

Would be good to add those remarks in coding guide.